docs/serving/services/configure-probing.md

```diff
@@ -0,0 +1,81 @@
+# Configuring Probing
+
+## General understanding of Knative Probing
+
+It is important to note that Knative probing is different from Kubernetes probing. 
+One reason for this is that Knative tries to minimize cold-start time and thus is probing
+in a vastly higher interval than Kubernetes does. 
+
+The general probing architecture looks like this:
+
+![probing-overview](./probes-overview.drawio.svg)
+
+* Users can optionally define Readiness and/or Liveness probes in the `KnativeService` CR.
+* The Liveness probes are directly executed by the Kubelet against the according container.
+* Readiness probes, on the other hand, are rewritten by Knative to be executed by the Queue-Proxy container to allow more aggressive probing and faster cold-start times.
+* Knative will define a default Readiness probe for the primary user container when no probe is defined by the user. It will check for a TCP socket on the traffic port of the Knative Service.
+* Knative will also define a Readiness probe for the Queue-Proxy container itself. Queue-Proxy's health endpoint aggregates all results from it's the rewritten Readiness probes for all user containers (primary + sidecars).
```

Comment:
Might be useful to describe a bit of the details like QP will hit each of the endpoints one by one and how the probe headers are passed. How the re-write looks like?